vis_sink.py

```diff
@@ -0,0 +1,374 @@
+        dest='hand_right',
+        default=False,
+        help='hand right side')
+    parser.add_option(
+        '--capture_dir',
+        dest='capture_dir',
+        default='./',
+        type='string',
+        help='capture directory')
+    parser.add_option(
+        '-c',
+        action='store_true',
+        dest='capture',
+        default=False,
+        help='Enable image capture')
+    parser.add_option(
+        '-a',
+        '--config_dir_arms',
+        dest='config_dir_arms',
+        default='robot_descriptions/tum-rosie/kinematics/lwr/',
+        type='string',
+        help='arms config data directory')
+    parser.add_option(
+        '-d',
+        '--config_dir_hands',
+        dest='config_dir_hands',
+        default='robot_descriptions/tum-rosie/kinematics/sahand/',
+        type='string',
+        help='hands config data directory')
+    (options, args) = parser.parse_args(argv[1:])
+    return (options, args)
+
+
+class VisSinkLoop(Controlloop):
+    def set_up(self):
+        self.current_time_for_log = 0
+
+        self.log = {}
+
+        self.options, args = configParse(sys.argv)
+
+        self.add_log({"id": "light0", "type": "Light", "pos": [10, 10, 10, 1]})
+        self.add_log({
+            "id": "light1",
+            "type": "Light",
+            "pos": [-10, 10, 10, 1]
+        })
+        self.add_log({"id": "light2", "type": "Light", "pos": [0, -10, 10, 1]})
+        self.add_log({"id": "light3", "type": "Light", "pos": [0, 0, -10, 1]})
+
+        self.add_log({
+            "id": "camera_center",
+            "type": "Disk",
+            "color": [0.5, 0.5, 0.5],
+            "color_reflex": ([1, 1, 1], 50),
+            "visibility": False
+        })
+
+        self.add_log({
+            "id": "world_frame",
+            "type": "Frame",
+            "scale": [0.3, 0.3, 0.3]
+        })
+
+        # All required info to get the robot descriptions (unpickable)
+        kin_desc = [
+            self.options, self.options.config_dir_arms,
+            self.options.config_dir_hands
+        ]
+
+        self.add_log({
+            "id": "arm_right",
+            "type": "Articulated",
+            "kinematic_description": kin_desc
+        })
+
+        self.add_log({
+            "id": "arm_left",
+            "type": "Articulated",
+            "kinematic_description": kin_desc
+        })
+
+        self.add_log({
+            "id": "hand_right",
+            "type": "Articulated",
+            "kinematic_description": kin_desc
+        })
+
+        self.add_log({
+            "id": "hand_left",
+            "type": "Articulated",
+            "kinematic_description": kin_desc
+        })
+
+        self.add_log({
+            "id":
+            "tree",
+            "type":
+            "Articulated_tree",
+            "children": ["arm_right", "arm_left", "hand_right", "hand_left"]
+        })
+
+        yarpbaseportname = '/' + self.options.robot + '/roboviewer'
```

Comment:
Please change this to `''.join(['/', self.options.robot ,'/roboviewer' ])`. Using `+` operators like this is discourage by PEP-8 and a bad practice [relevant docs](https://www.python.org/dev/peps/pep-0008/#programming-recommendations).

Reply:
Interesting, I did't know there was a performance penalty in other implementations. I find really difficult we will use any of those implementations, but will fix this anyways.

Reply:
Yes, and it unlikely this code will work on other implementations since we use cpython dependant APIs to interface with C in most of our code base.  However qouting PEP-8:

>  This optimization is fragile even in CPython
`join` is less fragile and will provide more consistent performance overall

Also, while performance is nice, I think the performance argument misses the point. It is also a style argument. using `join` is the preferred way of doing this according to the style and coding  standards we chose to follow.